In [123]:
import pandas as pd
filename = "/content/sales_predictions_2023.csv"
df = pd.read_csv(filename)
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


##Pregunta 1
8523 filas y 12 columnas

##Pregunta 2
Son Objetos, Flotantes y un Entero

In [124]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


##Pregunta 3
al parecer, no habrian duplicados

In [125]:
df = df.drop_duplicates()
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 8523 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8523 non-null   object 
 1   Item_Weight                7060 non-null   float64
 2   Item_Fat_Content           8523 non-null   object 
 3   Item_Visibility            8523 non-null   float64
 4   Item_Type                  8523 non-null   object 
 5   Item_MRP                   8523 non-null   float64
 6   Outlet_Identifier          8523 non-null   object 
 7   Outlet_Establishment_Year  8523 non-null   int64  
 8   Outlet_Size                6113 non-null   object 
 9   Outlet_Location_Type       8523 non-null   object 
 10  Outlet_Type                8523 non-null   object 
 11  Item_Outlet_Sales          8523 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 799.2+ KB


##Pregunta 4
Faltan 1463 datos en la columna Item_Weight y 2410 datos en la columna Outlet_Size

In [126]:
df.isna().sum()

Item_Identifier                 0
Item_Weight                  1463
Item_Fat_Content                0
Item_Visibility                 0
Item_Type                       0
Item_MRP                        0
Outlet_Identifier               0
Outlet_Establishment_Year       0
Outlet_Size                  2410
Outlet_Location_Type            0
Outlet_Type                     0
Item_Outlet_Sales               0
dtype: int64

#Pregunta 5
Para el caso de Item_Weight, calcule un promedio por codigo identificador (que deberia ser unico para cada producto) y luego rellene los valores vacios con el promedio por identificador, lo que restas son 4 registros con NaN, por lo que elimine esos 4 registros.

Para el caso del Outlet_Size
realizo un groupby de Outlet_Type con Outlet_Size para ver el cruce de informacion. Luego, sumo los valores NA por Outlet_Type cruzaod con Outlet_Size para comparar con el groupby realizado anteriormente, con esto me percato de que toda Grocery Store es Outlet_Size Small, por lo que procedo a rellenar los nulos entre ambos cruces (Grocery Store = Small) Luego de realizar varios cruces mas por las columnas de Outlet, me di cuenta que los datos faltantes pertenecen a Tier 2 de Outlet_Location_Type, por lo que rellene los valores con "Small" ya que es el valor que mas se repite en Tier 2 con valores.

In [127]:
#Primero Item_Weight.

promedio = df.groupby('Item_Identifier')['Item_Weight'].transform('mean')
df['Item_Weight'].fillna(promedio, inplace=True)

#con esto, ya quedan solo 4 datos sin dato en Item_Weight, y dado que es una muestra muy chica, los borro
df.dropna(subset=['Item_Weight'], inplace=True)

df["Outlet_Identifier"].value_counts()

df.groupby('Outlet_Identifier')['Outlet_Size'].value_counts()

df.loc[(df["Outlet_Size"].isnull()) & (df["Outlet_Type"] == "Grocery Store"), "Outlet_Size"] = "Small"

df.loc[(df["Outlet_Size"].isnull()) & (df["Outlet_Location_Type"] == "Tier 2"), "Outlet_Size"] = "Small"

df.groupby(["Outlet_Location_Type"] )["Outlet_Size"].apply(lambda x: x.isnull().sum())
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 8519 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8519 non-null   object 
 1   Item_Weight                8519 non-null   float64
 2   Item_Fat_Content           8519 non-null   object 
 3   Item_Visibility            8519 non-null   float64
 4   Item_Type                  8519 non-null   object 
 5   Item_MRP                   8519 non-null   float64
 6   Outlet_Identifier          8519 non-null   object 
 7   Outlet_Establishment_Year  8519 non-null   int64  
 8   Outlet_Size                8519 non-null   object 
 9   Outlet_Location_Type       8519 non-null   object 
 10  Outlet_Type                8519 non-null   object 
 11  Item_Outlet_Sales          8519 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 865.2+ KB


In [128]:
df.groupby(["Outlet_Location_Type"])['Outlet_Size'].value_counts()

Outlet_Location_Type  Outlet_Size
Tier 1                Small          1457
                      Medium          930
Tier 2                Small          2785
Tier 3                Medium         1860
                      High            932
                      Small           555
Name: count, dtype: int64

##Pregunta 6

In [129]:
df.isna().sum()

Item_Identifier              0
Item_Weight                  0
Item_Fat_Content             0
Item_Visibility              0
Item_Type                    0
Item_MRP                     0
Outlet_Identifier            0
Outlet_Establishment_Year    0
Outlet_Size                  0
Outlet_Location_Type         0
Outlet_Type                  0
Item_Outlet_Sales            0
dtype: int64

##Pregunta 7


In [130]:
df["Item_Fat_Content"].value_counts() #contiene errores Tiene low fat, LF y reg malos, deberian ser Low Fat o Regular
#df["Item_Type"].value_counts()
#df["Item_MRP"].value_counts()
#df["Outlet_Establishment_Year"].value_counts()
#df["Outlet_Size"].value_counts()
#df["Outlet_Location_Type"].value_counts()
#df["Outlet_Type"].value_counts()
#df["Item_Outlet_Sales"].value_counts()
#Procedo a corregir valores en Item_Fat_Content que es lo unico extraño que encontre
df.loc[df["Item_Fat_Content"] == "low fat", "Item_Fat_Content"] = "Low Fat"
df.loc[df["Item_Fat_Content"] == "reg", "Item_Fat_Content"] = "Regular"
df.loc[df["Item_Fat_Content"] == "LF", "Item_Fat_Content"] = "Low Fat"
df["Item_Fat_Content"].value_counts()
df.info()

<class 'pandas.core.frame.DataFrame'>
Index: 8519 entries, 0 to 8522
Data columns (total 12 columns):
 #   Column                     Non-Null Count  Dtype  
---  ------                     --------------  -----  
 0   Item_Identifier            8519 non-null   object 
 1   Item_Weight                8519 non-null   float64
 2   Item_Fat_Content           8519 non-null   object 
 3   Item_Visibility            8519 non-null   float64
 4   Item_Type                  8519 non-null   object 
 5   Item_MRP                   8519 non-null   float64
 6   Outlet_Identifier          8519 non-null   object 
 7   Outlet_Establishment_Year  8519 non-null   int64  
 8   Outlet_Size                8519 non-null   object 
 9   Outlet_Location_Type       8519 non-null   object 
 10  Outlet_Type                8519 non-null   object 
 11  Item_Outlet_Sales          8519 non-null   float64
dtypes: float64(4), int64(1), object(7)
memory usage: 865.2+ KB


##Pregunta 8

In [131]:
df.describe()

Item_Weight  Item_Visibility     Item_MRP  Outlet_Establishment_Year  \
count  8519.000000      8519.000000  8519.000000                8519.000000   
mean     12.875420         0.066112   141.010019                1997.837892   
std       4.646098         0.051586    62.283594                   8.369105   
min       4.555000         0.000000    31.290000                1985.000000   
25%       8.785000         0.026983    93.844900                1987.000000   
50%      12.650000         0.053925   143.047000                1999.000000   
75%      16.850000         0.094558   185.676600                2004.000000   
max      21.350000         0.328391   266.888400                2009.000000   

       Item_Outlet_Sales  
count        8519.000000  
mean         2181.188779  
std          1706.511093  
min            33.290000  
25%           834.247400  
50%          1794.331000  
75%          3100.630600  
max         13086.964800